### ISSUE TITLE
bug: .at is not a function

### ISSUE THREAD

**Issue by AntoineAwaida** [ionic-team/capacitor#7154]
### Capacitor Version

5.5.1

```
💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 5.6.0
  @capacitor/core: 5.6.0
  @capacitor/android: 5.6.0
  @capacitor/ios: 5.6.0

Installed Dependencies:

  @capacitor/cli: 5.5.1
  @capacitor/android: 5.5.1
  @capacitor/ios: 5.5.1
  @capacitor/core: 5.5.1

[success] iOS looking great! 👌
[success] Android looking great! 👌
```

### Platform(s)

iOS pre 16

### Current Behavior

iOS app crashes when using the `.at` javascript function introduced in ES2022. This only happens on iOS versions <16.

<img width="771" alt="Screenshot 2023-12-22 at 14 55 21" src="https://github.com/ionic-team/capacitor/assets/17647473/1dd6b295-2a8f-4026-9045-c7b00275f208">

### Expected Behavior

Should not raise an error (as my compilerOption target in tsconfig.ts is ES2020). I tried with ES2015 as well.

### Code Reproduction

- Create a capacitor project 
- Right a function using the `.at` method (for instance : elements.at(0)).
- Run the application on an iOS platform with an installed version <16.

### Other Technical Details

`npm --version` output:
9.8.1

`node --version` output:
18.18.0

`pod --version` output (iOS issues only):
1.13.0

iOS version : 
14.7.1

### Additional Context

Does the Capacitor requirement I've read on the Internet for a current or previous version of iOS still stands ?

**Comment by rawrmonstar** [comment]
I am pretty certain this isn't a bug in capacitor. You need to include a polyfill to support features such as `Array.prototype.at` in older Web Views like the ones before iOS 16. Typescript's transpilation does not include polyfills, the target is about the syntax it produces.

Capacitor's supported platforms is about Capacitor being able to run on those platforms, not that it will transform your JS code to be compatible with those older platforms.

**Comment by AntoineAwaida** [comment]
Thanks for your reply. Actually that's what I figured out later on this afternoon.

However it might be still relevant for other users. What I struggled with was that I was building my code with a ES2020 target, but .at methods were still kept in build outputs (this was introduced in ES2022).
**It turns out that .at is kept because it is interpreted (in my case at least) as a Node method provided for RelativeIndexables, even though I am building code to be used in a browser !**

See related issue : https://stackoverflow.com/questions/70019324/how-does-create-react-app-allow-for-array-prototype-at-while-its-not-allowed

Very concerning in my opinion. Especially as there's no way to tell Typescript/ESLint to not import Node types at the very least.
